dotnet/src/Microsoft.AutoGen/Agents/Agent.cs

```diff
@@ -12,30 +13,30 @@
 
 namespace Microsoft.AutoGen.Agents;
 
-public abstract class AgentBase : IAgentBase, IHandle
+public abstract class Agent : IDisposable, IHandle
 {
     public static readonly ActivitySource s_source = new("AutoGen.Agent");
     public AgentId AgentId => _runtime.AgentId;
     private readonly object _lock = new();
-    private readonly Dictionary<string, TaskCompletionSource<RpcResponse>> _pendingRequests = [];
+    private readonly ConcurrentDictionary<string, TaskCompletionSource<RpcResponse>> _pendingRequests = [];
 
     private readonly Channel<object> _mailbox = Channel.CreateUnbounded<object>();
     private readonly IAgentRuntime _runtime;
     public string Route { get; set; } = "base";
```

Comment:
This `Route` is not used anywhere, maybe remove it?

Reply:
I think we can remove it.